sdk/python/kfp/v2/components/types/type_annotations.py

```diff
@@ -131,3 +142,49 @@ def get_io_artifact_annotation(typ):
         return subs_tree[2]
 
     return typ.__args__[1]
+
+
+def maybe_strip_optional_from_annotation(annotation: T) -> T:
```

Comment:
This is copied over from https://github.com/kubeflow/pipelines/blob/2f19a26ffd205c7a3384c4c250f5dbb5150a7c4d/sdk/python/kfp/components/type_annotation_utils.py#L22